datacats/cli/deploy.py

```diff
@@ -28,8 +27,8 @@ def deploy(environment, opts):
 TARGET_NAME is the name of the environment on DataCats.com. Defaults to
 the environment name.
 """
-    profile = get_working_profile(environment)
-
+    if not profile:
+        raise Exception("Need to pass a loaded user_profile instance")
```

Comment:
In general it's bad to raise Exception. Wouldn't this fire if the above function call fails, not if user_profile wasn't passed?  If this will only happen if we did something wrong, I wouldn't bother checking here, just let it fail later on instead.